gender_novels/analysis/instance_distance_analysis.py

```diff
@@ -1,47 +1,231 @@
+    date_1880_to_1889 = []
+    date_1890_to_1899 = []
+    date_1900_on = []
+
+    for k in list(results.keys()):
+        if k.date < 1810:
+            date_to_1810.append(results[k][2][stat])
+        elif k.date < 1820:
+            date_1810_to_1819.append(results[k][2][stat])
+        elif k.date < 1830:
+            date_1820_to_1829.append(results[k][2][stat])
+        elif k.date < 1840:
+            date_1830_to_1839.append(results[k][2][stat])
+        elif k.date < 1850:
+            date_1840_to_1849.append(results[k][2][stat])
+        elif k.date < 1860:
+            date_1850_to_1859.append(results[k][2][stat])
+        elif k.date < 1870:
+            date_1860_to_1869.append(results[k][2][stat])
+        elif k.date < 1880:
+            date_1870_to_1879.append(results[k][2][stat])
+        elif k.date < 1890:
+            date_1880_to_1889.append(results[k][2][stat])
+        elif k.date < 1900:
+            date_1890_to_1899.append(results[k][2][stat])
+        else:
+            date_1900_on.append(results[k][2][stat])
+
+    data['date_to_1810'] = date_to_1810
+    data['date_1810_to_1819'] = date_1810_to_1819
+    data['date_1820_to_1829'] = date_1820_to_1829
+    data['date_1830_to_1839'] = date_1830_to_1839
+    data['date_1840_to_1849'] = date_1840_to_1849
+    data['date_1850_to_1859'] = date_1850_to_1859
+    data['date_1860_to_1869'] = date_1860_to_1869
+    data['date_1870_to_1879'] = date_1870_to_1879
+    data['date_1880_to_1889'] = date_1880_to_1889
+    data['date_1890_to_1899'] = date_1890_to_1899
+    data['date_1900_on'] = date_1900_on
+
+    return data
+
+def results_by_location(results, metric):
+    """
+
+    :param results:
+    :param metric:
+    :return:
+    """
+    data = {}
+    metric_indexes = {"median": 0, "mean": 2, "min": 3, "max": 4}
+    try:
+        stat = metric_indexes[metric]
+    except:
+        print("Not valid metric name. Valid names: 'median', 'mean', 'min', 'max'")
+
+    location_England = []
+    location_US = []
+    location_other = []
+
+    for k in list(results.keys()):
+        if k.country_publication == 'England':
+            location_England.append(results[k][2][stat])
+        elif k.country_publication == 'United States':
+            location_US.append(results[k][2][stat])
         else:
-            count += 1
-            if element == word:
-                output.append(count)
-                count = 0
-    return output
+            location_other.append(results[k][2][stat])
+
+    data = {}
+
+    data['location_England'] = location_England
+    data['location_US'] = location_US
+    data['location_other'] = location_other
+
+    return data
+
+def run_gutenberg_analysis():
+    print("loading corpus")
+    corpus = Corpus('gutenberg')
+    novels = corpus.novels
+
+    print("running analysis")
+    results = run_distance_analysis(novels)
+
+    print("storing results")
+    store_raw_results(results, "gutenberg")
+
+if __name__ == '__main__':
```

Comment:
There shouldn't be any new functions defined below name=main.

Reply:
Ah sorry, that's a mistake, just some functions that I meant to delete. Will get rid of them.